ext/src/ruby_api/store.rs

```diff
@@ -390,10 +429,30 @@ impl ResourceLimiter for TrackingResourceLimiter {
         desired: usize,
         maximum: Option<usize>,
     ) -> anyhow::Result<bool> {
+        if let Some(max) = maximum {
+            if desired > max {
+                self.linear_memory_limit_hit = true;
+                return Ok(false);
+            }
+        }
```

Comment:
This check is already performed by the default implementation, (the `self.inner.memory_growing` call below) perhaps we can omit this, in general I'd say that it might make sense to leave the body of this function as:

```rust
if res.is_ok() {
    self.tracker.increase_memory_usage(desired - current);
 } else {
     self.linear_memory_limit_hit = true;
  }

   if let Ok(allowed) = res {
      if !allowed {
         self.linear_memory_limit_hit = true;
       }
   }
```